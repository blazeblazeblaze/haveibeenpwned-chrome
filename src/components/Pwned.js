import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';
import RestartLink from './RestartLink';

const Pwned = ({onStartOver, email}) => {
  return (
    <div>
      <p>
        <strong>Oh no!</strong> It looks like {email} has been pwned!
      </p>
      <p>
        <Link
          className='pure-button pure-button-primary button-small'
          href={`https://haveibeenpwned.com/account/${email}`}>
          Read more
        </Link>
        {' '}
        <RestartLink onStartOver={onStartOver}/>
      </p>
    </div>
  );
}

Pwned.propTypes = {
  onStartOver: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired
};

export default Pwned;
