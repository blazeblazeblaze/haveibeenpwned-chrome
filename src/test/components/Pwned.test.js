import React from 'react';
import ReactDOM from 'react-dom';
import Pwned from '../../components/Pwned';
import renderer from 'react-test-renderer';

const email = 'foo@example.com';
jest.mock('../../components/Link', () => () => 'Link');
jest.mock('../../components/RestartLink', () => () => 'RestartLink');

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Pwned onStartOver={jest.fn } email={email} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Pwned renders sad text and links', () => {
  const component = renderer.create(
    <Pwned onStartOver={jest.fn} email={email} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
