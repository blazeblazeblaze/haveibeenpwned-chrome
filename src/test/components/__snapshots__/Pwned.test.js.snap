// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pwned renders sad text and links 1`] = `
<div>
  <p>
    <strong>
      Oh no!
    </strong>
     It looks like 
    foo@example.com
     has been pwned!
  </p>
  <p>
    Link
     
    RestartLink
  </p>
</div>
`;
